src/AzureExtension/DevBox/DevBoxProvider.cs

```diff
@@ -143,6 +143,15 @@ public async Task<IEnumerable<IComputeSystem>> GetDevBoxesAsync(IDeveloperId dev
         // update the cache every time we retrieve new Dev Boxes. This is used so in the creation flow we don't need
         // to retrieve the Dev Boxes again if the user already retrieved them in the environments page in Dev Home
         _cachedDevBoxesMap[GetUniqueDeveloperId(developerId)] = devBoxes.ToList();
+
+        // Get Associated pools the first time we get the projects
+        var uniqueUserId = GetUniqueDeveloperId(developerId);

```

Comment:
we can move this to line 145 and then use uniqueUserId  for the key in line 145

Reply:
And line 147 :)